### DISCUSSION TITLE
Should the inactive entities being taken into account into the system of equations?

### DISCUSSION THREAD

**Discussion by loumalouomega** [KratosMultiphysics/Kratos#12899] [General]
When constructing the graph, inactive elements/conditions/MPCs are taken into account, even though they do not participate in the system of equations. I understand that this is intentional, allowing for the activation/deactivation of these elements without needing to reform the degrees of freedom (DoF) at each step. However, this approach can be problematic, as it assigns a value of 1 to the diagonal, potentially introducing a constraint that could conflict with shared nodes between inactive and active entities.

**Comment by loumalouomega** [comment]
See https://github.com/KratosMultiphysics/Kratos/blob/85e11e6bbbeadb45d3fa9899f0450955c8d39126/kratos/utilities/dof_utilities/block_build_dof_array_utility.h#L109

**Comment by RiccardoRossi** [comment]
in principle it should not be the case. The 1 is added only if the row is empty...am i wrong?

Riccardo

**Reply by loumalouomega** [reply]
That's what i thougth, but it happens

**Comment by RiccardoRossi** [comment]
in any case, all of the part which is related to MPCs (better said, Master Slave Constraints) should be completely rewritten ...mpcs should not be included in the dofs of the system

**Reply by loumalouomega** [reply]
First, I would try not to speak about MPC because it is not the subject of this discussion. Second, if a MPC has a dof that is not part of the elements or conditions will not be included and crash.
